presto-main/src/main/java/com/facebook/presto/sql/planner/SortExpressionExtractor.java

```diff
@@ -152,26 +206,59 @@ public boolean equals(Object obj)
                 return false;
             }
             SortExpression other = (SortExpression) obj;
-            return Objects.equals(this.channel, other.channel);
+            return Objects.equals(channel, other.channel) &&
+                Objects.equals(filters, other.filters);
         }
 
         @Override
         public int hashCode()
         {
-            return Objects.hash(channel);
+            return Objects.hash(channel, filters);
         }
 
         public String toString()
         {
             return toStringHelper(this)
                     .add("channel", channel)
+                    .add("filters", filters)
                     .toString();
         }
 
-        public static SortExpression fromExpression(Expression expression)
+        public static int fieldReferenceIndex(Expression expression)
         {
             checkState(expression instanceof FieldReference, "Unsupported expression type [%s]", expression);
-            return new SortExpression(((FieldReference) expression).getFieldIndex());
+            return ((FieldReference) expression).getFieldIndex();
+        }
+
+        public static List<Expression> filters(Expression expression)
+        {
+           return new FilterExpressions().process(expression);
+        }
+
+        private static class FilterExpressions
+                extends AstVisitor<List<Expression>, Void>
+        {
+            @Override
+            protected List<Expression> visitExpression(Expression expression, Void context)
+            {
+                List<Expression> filters = new ArrayList<>();
+                for (Node child : expression.getChildren()) {
+                    filters.addAll(process(child));
+                }
+                return filters;
+            }
+
+            @Override
+            protected List<Expression> visitLogicalBinaryExpression(LogicalBinaryExpression expression, Void context)
+            {
+                return extractConjuncts(expression);
```

Comment:
This does not seem good.
It will work correctly for case like this
`a < x AND a < x +10`

But what if we have other conjuncts in filter function. Which are not in shape of
`[sort_symbol]  [<=|>=|<|>] [expression using probe side symbols>]`.

Then those conjuncts will be used as filter functions in `SortedPositionLInks.start()`. And they will not work fine when passed as `lessThanFunction` to binary search.